edit-post/components/modes/visual-editor/index.js

```diff
@@ -1,35 +1,32 @@
-/**
- * External dependencies
- */
-import { connect } from 'react-redux';
-
 /**
  * WordPress dependencies
  */
 import {
-	BlockList,
+	BlockListLayout,
 	PostTitle,
 	WritingFlow,
 	EditorGlobalKeyboardShortcuts,
 	BlockSelectionClearer,
 } from '@wordpress/editor';
 import { Fragment } from '@wordpress/element';
+import { query } from '@wordpress/data';
 
 /**
  * Internal dependencies
  */
 import './style.scss';
 import BlockInspectorButton from './block-inspector-button';
-import { hasFixedToolbar } from '../../../store/selectors';
 
-function VisualEditor( props ) {
+function VisualEditor( { blockUIDs, hasFixedToolbar } ) {
 	return (
 		<BlockSelectionClearer className="edit-post-visual-editor">
 			<EditorGlobalKeyboardShortcuts />
 			<WritingFlow>
 				<PostTitle />
-				<BlockList
-					showContextualToolbar={ ! props.hasFixedToolbar }
+				<BlockListLayout
```

Comment:
Should we push the refactor a little further, and assume that if there's no prop, we should render the default layout with the root blocks? 

This would have the advantage of avoiding the exposed `getBlockOrder` selector.